### DISCUSSION TITLE
how to get traits value

### DISCUSSION THREAD

**Discussion by jaspreet-singh-1999** [GrapesJS/grapesjs#5577] [Q&A] [answered]
hi,
i want to get present component trait values. how can i do

**Comment by mosh-tudor** [comment]
Try `component.getAttributes()` or `component.model.getAttributes()` _(I don't remember which of them works)_

**Comment by artf** [comment, marked as answer]
A trait could be assigned to an attribute or a property of a component (if `changeProp` is used), so it's better to get the value directly from the trait
```js
const trait = component.getTrait('TRAIT_NAME');
console.log('trait value', trait.getValue());
```
